### PR TITLE
docs: Explain where native binary currently ends up (re. #508)

### PR DESCRIPTION
**Pull request by vorburger**
see #508

This would be updated to document `-o/--output` when that's implemented.

I'll see if I can have a go at that some time.

### CODE REVIEW

**Comment by vorburger** [comment]
This PR has 2 commits because its based on / includes my earlier #507, just to avoid having to resolve a merge conflict when that's in.

File: readme.adoc [hunk 1]
```diff
@@ -391,9 +391,14 @@ There is support for using `native-image` from GraalVM project to produce a bina
 
 Since not all java libraries can automatically be built with `native-image` - especially if using reflection feature are considered highly experimental.
 
-Just run `jbang --native helloworld.java` and `jbang` will use `native-image` from either `JAVA_HOME/bin` or `GRAALVM_HOME/bin` to
+Just run `jbang --native helloworld.java` and `jbang` will use `native-image` from either `$JAVA_HOME/bin` or `$GRAALVM_HOME/bin` or `$PATH` to
 produce a native image binary.
 
+The built native image binary will be in `~/.jbang/cache/jars/YOUR.java.HASH.jar.bin` (see [#508] for an `-o/--output`).
```
Comment (maxandersen) [inline readme.adoc:397]:
```suggestion
The built native image binary will be placed in jbang's cache and used on next run.
```

I don't want to make the contents of ~/.jbang/cache "api" ...at leats for now it is to be considered a black box for users.

Definitely ok adding a `-o/--out` to make this possible to get..

**Review by maxandersen — CHANGES_REQUESTED** [review]
lets see if we can find phrasing not making cache content documented.

**Comment by maxandersen** [comment]
merging with a hint rather than documenting internal details.

**Pull request closed without merging**